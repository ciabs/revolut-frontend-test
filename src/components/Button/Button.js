import React from 'react';
import PropTypes from 'prop-types';

import {Container} from './Button.styles';

const Button = ({isDisabled, onClick, children}) => {

  return (
    <Container
      disabled={isDisabled}
      onClick={onClick}
    >
      {children}
    </Container>
  );
};

Button.propTypes = {
  isDisabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Button;
