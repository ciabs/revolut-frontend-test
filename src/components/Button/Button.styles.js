import styled from 'styled-components';

import {
  EXCHANGE_BG_COLOR_DISABLED,
  EXCHANGE_BG_COLOR_ENABLED, EXCHANGE_BG_COLOR_HOVER, EXCHANGE_COLOR
} from '../../styles/constants';

export const Container = styled.button`
  cursor: pointer;
  background-color: ${EXCHANGE_BG_COLOR_ENABLED};
  color: ${EXCHANGE_COLOR};
  border: none;
  width: 80%;
  padding: 15px;
  border-radius: 25px;
  box-shadow: 0 10px 10px 0 rgba(0,0,0,0.15);
  outline: none;
  position: absolute;
  bottom: 20px;
  left: 50%;
  transform: translateX(-50%);
  :hover {
    background-color: ${EXCHANGE_BG_COLOR_HOVER};
  }
  &:disabled {
    background-color: ${EXCHANGE_BG_COLOR_DISABLED};
    cursor: auto;
  }
`;
