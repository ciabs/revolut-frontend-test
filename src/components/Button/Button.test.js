import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  const props = {
    isDisabled: false,
    onClick: jest.fn(),
    children: 'hello world'
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Button {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Button {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
