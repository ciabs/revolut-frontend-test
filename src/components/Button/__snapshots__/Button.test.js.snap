// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="sc-bdVaJa cgzLHg"
  disabled={false}
  onClick={[Function]}
>
  hello world
</button>
`;
